content/docs/rfcs/38/README.md

```diff
@@ -0,0 +1,480 @@
+of a shared, trusted state machine evolution.  Such a state machine
+may perform an arbitrary computation from the class of "smart
+contract" artifacts by implementing a
+suitable model of transaction.  
+
+# Glacier 
+
+The Glacier consensus algorithm computes a yes/no decision via a set
+of distributed computational nodes.  Glacier is a leaderless
+probabilistic binary consensus algorithm with fast finality that
+provides good reliability for network and Byzantine fault tolerance.
+
+## Algorithm 
+
+Given a set of $n$ distributed computational nodes, a gossip-based
+protocol is presumed to exist which which allows members to discover,
+join, and leave a possibly maximally connected graph.  Joining this
+graph allows each node to view a possibly incomplete node membership
+list of all other nodes.  This view may change as the protocol
+advances, as nodes join and leave.  
+
+
+### Pre-requisites
+
+A proposal is formulated to which consensus of truth or falsity is
+desired.  Each node that participates starts the protocol with an
+opinion on the proposal, represented in the sequel as **YES**, **NO**,
+and **UNDECIDED**.
+
+     opinon
+        <-- choose local truth computation of  { YES, NO, UNDECIDED}
+    
+The algorithm proceeds in rounds for each node.  The liveness of the
+algorithm is severely constrained in the absence of timeouts for a
+round to proceed.
+
+### Setup
+
+The node initializes the following constants variables 
+
+    ;;; The following values are constants chosen with justification from experiments
+    ;;; performed with the adversarial models
+    
+    ;; FIXME: give an empiral confidence threshold 
+    confidence_threshold
+     <-- T  ;; BOGUS:  this should be a function of the network size
+            ;; and the current confidence in the observer majority
+    
+    ;; constant look ahead parameter, 
+    look_ahead 
+      <-- 20
+    
+    ;; first order confidence smoothing parameter
+    $alpha_1$ 
+      <-- 0.8  ;; empirically derived from consensus simulation
+    
+    ;; second order confidence smoothing parameter
+    $alpha_2$ 
+      <-- 0.4  ;; empirically derived from consensus simulation
+    
+    ;;; The following variables are initialized
+    
+    ;; number of nodes to uniformly randomly query  
+    k 
+      <-- 7
+
+    ;; neighbor threshold multiplier
+    k_multiplier 
+      <-- 2
+
+    ;; maximal threshold multiplier, i.e. we will never exceed 
+    ;; questioning k_multiplier * max_k_multiplier peers
+    max_k_multiplier 
+      <-- 4
+      
+    ;; total number of votes
+    total_votes 
+      <-- 0    
+    ;; total number of positive nodes
+    total_positive 
+       <-- 0
+
+###  Query 
+
+A node selects $k$ nodes randomly from the locally known complete set
+of peers in the network.
+
+Each node replies with their current opinion on the proposal.
+
+When the query finishes, the node now initializes the following two
+values:
+
+ 
+    new_votes 
```

Comment:
Is this just $k$ when all nodes are available?

Reply:
Not all votes are necessarily collected in a round (time outs).

Reply:
Ok